/*
Copyright 2017-2018 Echo Park Labs

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

For additional information, contact:

email: info@echoparklabs.io
*/

syntax = "proto2";

option java_multiple_files = true;
option java_package = "com.epl.protobuf.stac";
option java_outer_classname = "StacMetadata";
option objc_class_prefix = "STC";

package epl.protobuf;

import "google/protobuf/timestamp.proto";
import "epl/protobuf/geometry_operators.proto";
import "epl/protobuf/stac.proto";


message EOAsset {
    required string href = 1;
    required STAC_ASSET_TYPE eo_cloud_cover = 2;
    optional EO_BAND eo_band = 3;
}

message MetadataResult {
    //    https://github.com/protocolbuffers/protobuf/issues/1606
    optional double eo_cloud_cover = 6;
    optional double eo_sun_azimuth = 7;
    optional double eo_sun_elevation = 8;
    optional double eo_gsd = 9;
    optional double eo_off_nadir = 10;
    // TODO really, these should be a float, but python shits the bed at the idea of a 4byte float.

    optional BBoxField eo_bbox = 11;
    optional epl.protobuf.GeometryBagData eo_geometry = 12;
    map<string, EOAsset> asset_map = 13;

    reserved 14 to 200;

    optional google.protobuf.Timestamp src_img_date = 201;
    optional string state_initials = 202;
    optional string filename = 203;
}


